### ISSUE TITLE
MultiList can't use customQuery with empty defaultValue

### ISSUE THREAD

**Issue by gauthierm** [appbaseio/reactivesearch#1402]
**Affected Projects**
React

**Library Version:** x.y.z
```
  "@appbaseio/reactivecore": "^9.0.3",
  "@appbaseio/reactivesearch": "^3.2.4",
```

**Describe the bug**
`MultiList` with a `customQuery` never calls custom query when default value is an empty array.

**To Reproduce**
```jsx
<MultiList
  dataField="type"
  defaultValue={[]}
  customQuery={(value, props) => {
    if (value.length) {
      return { query: { terms: { [props.dataField]: value } } };
    }

    return { query: { bool: must_not: { term: { [props.dataField]: 'comment' } } };
  }}
/>
```

**Expected behavior**
Default query excludes type = comment documents.

**Additional context**
This is caused by the MultiList component constructor not calling setValue and thus not calling updateQuery when the default value is empty.

**Comment by gauthierm** [comment]
See https://github.com/appbaseio/reactivesearch/blob/next/packages/web/src/components/list/MultiList.js#L95-L97

**Comment by gauthierm** [comment]
Because the base MultiList is not exported it's difficult to work around this issue. I was able to do so by setting a hacky default value:

```js
const defaultDocumentType = { forEach: () => {}, length: 1, isFake: true };
```

**Comment by crab86** [comment]
Hi @gauthierm : Could you describe your hacky solution in detail? I am new to react/javascript but nevertheless i am using MultiList with defaultQuery and customQuery as well and as soon as i am clearing the MultiList filter via Selected Filters component - my Multilist vanishes. Maybe this is related so i would really like to see your hack!

**Comment by ShahAnuj2610** [comment]
Right. Initially the query should have been with called with `comment` as value.

Reproducible CSB: https://codesandbox.io/s/wizardly-pascal-zim38?file=/src/index.js

**Comment by ShahAnuj2610** [comment]
@gauthierm I think it's not a bug. **customQuery** only gets applied when a `value` changes in `MultiList` components.

![](https://i.imgur.com/Dbyrtsh.png)

 So if you want your query with value `comment` applied, then you can pass it as a defaultQuery in `MultiList` component as I have done here: https://codesandbox.io/s/stupefied-babbage-pyvo6?file=/src/index.js. Lmk if you have any further doubts.

Ref: https://docs.appbase.io/docs/reactivesearch/v3/list/multilist/#props

cc: @bietkul

**Comment by gauthierm** [comment]
@ShahAnuj2610 The `defaultQuery` affects the query for values displayed in the multilist and the `customQuery` affects the query in the parent reactive base.

Your suggestion to use `defaultQuery` just makes my default inverse value disappear from the multilist and does not affect the main search query.

The `customQuery` and `defaultQuery` documentation in the appbase docs is not particularly clear and the props names themselves seem misleading but reading the implementation this seems the be the intent. See also https://github.com/appbaseio/reactivesearch/pull/954 and https://github.com/appbaseio/reactivesearch/issues/855
